EventView-1/src/main/java/com/eventview/repo/EventRepoImpl.java

```diff
@@ -22,33 +25,81 @@
+        try {
+            EventsPayload eventsPayload = jdbcTemplate.queryForObject(SELECT_EVENT_BY_ID_CUS, new Object[]{eventId}, new EventsPayloadRowMapper());
+            log.info("query generated " + SELECT_EVENT_BY_ID_CUS + "-----" + eventId);
+            return eventsPayload;
+        } catch (Exception e) {
+            throw new EventNotFoundException("Event not found with id: " + eventId);
+        }
     }
 
     @Override
     public void createEvent(Events event) {
-        String ce = "INSERT INTO events (event_id, user_id,event_type_id, event_date) VALUES (?,?,?,?)";
-        jdbcTemplate.update(ce,event.getEventId(),event.getUserId(), event.getEventTypeId(), event.getEventdate());
+        String INSERT_EVENT = "INSERT INTO events (event_id, user_id,event_type_id, event_date) VALUES (?,?,?,?)";
+        jdbcTemplate.update(INSERT_EVENT, event.getEventId(), event.getUserId(), event.getEventTypeId(), event.getEventDate());
     }
 
     @Override
     public void updateEvent(Events event) {
-        jdbcTemplate.update("update events set user_id=?,event_type_id=?, event_date=? where event_id=?",
-                event.getUserId(),event.getEventTypeId(),event.getEventdate(),event.getEventId());
+        String UPDATE_EVENT = "update events set user_id=?,event_type_id=?, event_date=? where event_id=?";
+        jdbcTemplate.update(UPDATE_EVENT,
+                event.getUserId(), event.getEventTypeId(), event.getEventDate(), event.getEventId());
+    }
+
+    @Override
+    public int deleteEvent(Integer eventId) {
+        String DELETE_EVENT = "delete from events where event_id=?";
+        int size = jdbcTemplate.update(DELETE_EVENT, eventId);
+        if(size==0) {
+            throw new EventNotFoundException("No Event found to delete: "+eventId);
+        }
+        return size;
+    }
+
+    @SuppressWarnings("ConstantConditions")
+    @Override
+    public boolean eventExists(Integer eventId) {
+        String EVENT_EXISTS = "SELECT count(*) FROM events WHERE event_id = ?";
+        int count = jdbcTemplate.queryForObject(EVENT_EXISTS, new Object[]{ eventId }, Integer.class);
+        return count > 0;
+    }
+
+    @Override
+    public List<Integer> getAllEventMonths() {
+        String SELECT_MONTHS = "SELECT month(event_date) from events where month(now()) = MONTH(event_date)";
+        return jdbcTemplate.queryForList(SELECT_MONTHS,Integer.class);
     }
 
     @Override
-    public void deleteEvent(Integer eventid) {
-        jdbcTemplate.update("delete from events where event_id=?", new Object[]{eventid}, new EventRowMapper());
-        System.out.println("Record with id:" + eventid + " are deleted");
+    public List<Integer> getAllEventDay() {
+        String SELECT_DAYS = "select day(event_date) from events where day(now()) = day(event_date)";
+        return jdbcTemplate.queryForList(SELECT_DAYS,Integer.class);
     }
```

Comment:
as mentioned in other comment this is not needed if u implement other way

Reply:
still this code exists?

Reply:
removed